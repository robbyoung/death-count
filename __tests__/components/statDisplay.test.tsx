import React from 'react';
import renderer from 'react-test-renderer';
import StatDisplay from '../../app/components/statDisplay';

describe('Stat Display Component', () => {
    it('can render a stat display', () => {
        const component = renderer.create(<StatDisplay
            count={2}
            name={'Test Stat Display'}
            percentage={20}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});