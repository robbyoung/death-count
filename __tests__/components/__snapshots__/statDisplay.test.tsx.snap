// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Display Component can render a stat display 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e84641",
      "borderRadius": 5,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "padding": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Test Stat Display
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "width": "50%",
        }
      }
    >
      2
       Deaths
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "textAlign": "right",
          "width": "50%",
        }
      }
    >
      20
      % of Deaths
    </Text>
  </View>
</View>
`;
